sockeye/train.py

```diff
@@ -79,27 +84,29 @@ def _dict_difference(dict1: Dict, dict2: Dict):
     return diffs
 
 
-def main():
-    params = argparse.ArgumentParser(description='CLI to train sockeye sequence-to-sequence models.')
-    arguments.add_train_cli_args(params)
-    args = params.parse_args()
+def seedRNGs(args: argparse.Namespace) -> None:
+    """
+    Seed the random number generators (Python, Numpy and MXNet)
 
-    # seed the RNGs
+    :param args: Arguments as returned by argparse.
+    """
     np.random.seed(args.seed)
     random.seed(args.seed)
     mx.random.seed(args.seed)
 
-    if args.use_fused_rnn:
-        check_condition(not args.use_cpu, "GPU required for FusedRNN cells")
 
-    check_condition(args.optimized_metric == C.BLEU or args.optimized_metric in args.metrics,
-                    "Must optimize either BLEU or one of tracked metrics (--metrics)")
+def check_resume(args: argparse.Namespace, output_folder: str) -> Tuple[bool, str]:
+    """
+    Check if we should resume a broken training run.
 
-    # Checking status of output folder, resumption, etc.
+    :param args: Arguments as returned by argparse.
+    :param output_folder: Main output folder for the model.
+    :return: Flag signaling if we are resuming training and the directory with
+        the training status
+    """
     # Create temporary logger to console only
     logger = setup_main_logger(__name__, file_logging=False, console=not args.quiet)
```

Comment:
why create the temporary logger here again? You added it outside of main at the top.
You might need to use global logger in the main then...

Reply:
Right! I forgot to remove it here!